Source/Request.swift

```diff
@@ -405,10 +405,42 @@ open class DownloadRequest: Request {
 
     // MARK: Helper Types
 
-    /// A closure executed once a request has successfully completed in order to determine where to move the temporary
-    /// file written to during the download process. The closure takes two arguments: the temporary file URL and the URL
-    /// response, and returns a single argument: the file URL where the temporary file should be moved.
-    public typealias DownloadFileDestination = (URL, HTTPURLResponse) -> URL
+    /// A collection of options to be executed prior to moving a downloaded file from the temporary URL to the
+    /// destination URL.
+    public struct DownloadOptions: OptionSet {
+        /// Defines the `RawValue` type as `UInt` to satisfy the `RawRepresentable` protocol.
+        public typealias RawValue = UInt
+
+        /// Returns the raw bitmask value of the option and satisfies the `RawRepresentable` protocol.
+        public let rawValue: RawValue
```

Comment:
You can just define `rawValue` with a type and it will be inferred. This can be simplified to:
```swift
public struct DownloadOptions: OptionSet {
    /// Returns the raw bitmask value of the option and satisfies the `RawRepresentable` protocol.
    public let rawValue: UInt
    
    /// A `DownloadOptions` flag that creates intermediate directories for the destination URL if specified.
    public static let createIntermediateDirectories = DownloadOptions(rawValue: 1 << 0)
    
    /// A `DownloadOptions` flag that removes a previous file from the destination URL if specified.
    public static let removePreviousFile = DownloadOptions(rawValue: 1 << 1)
    
    // MARK: Initialization Methods
    
    /// Creates a `DownloadFileDestinationOptions` instance with the specified raw value.
    ///
    /// - parameter rawValue: The raw bitmask value for the option.
    ///
    /// - returns: A new log level instance.
    public init(rawValue: UInt) { self.rawValue = rawValue }
}
```
Weirdly the init can't be generated automatically for some reason. But the intermediate static values seem unnecessary, unless there's something I'm missing?